pkg/agent/variables.go

```diff
@@ -178,6 +178,10 @@ func getOutBoundCmd(nbc *datamodel.NodeBootstrappingConfiguration, cloudSpecConf
 		return ""
 	}
 
+	if nbc.OutboundType == "none" || nbc.OutboundType == "block" {
```

Comment:
string is fine but wondering if it would be better to use an enum type or something, e.g.

package datamodel

...

type OutboundType int

const (
    OutboundTypeNone = iota
    OutboundTypeBlock
)

Reply:
actually using string is easier since we can more easily denote RP having not set it at all, so I'm fine with that

Reply:
though you could make it work by having the unspecified type be the initial `iota`, such as:

https://github.com/Azure/AgentBaker/blob/master/pkg/agent/datamodel/types.go#L1680-L1686